pkg/gameserverallocations/controller.go

```diff
@@ -493,44 +509,154 @@ func (c *Controller) serialisation(r *http.Request, w http.ResponseWriter, obj k
 	return errors.Wrapf(err, "error encoding %T", obj)
 }
 
-// allocate allocated a GameServer from a given Fleet
+// allocate allocated a GameServer from a given GameServerAllocation
+// this sets up allocation through a batch process.
 func (c *Controller) allocate(gsa *v1alpha1.GameServerAllocation) (*stablev1alpha1.GameServer, error) {
-	var allocation *stablev1alpha1.GameServer
-	var comparator findComparator
+	req := request{gsa: gsa, response: make(chan response)}
 
-	switch gsa.Spec.Scheduling {
-	case apis.Packed:
-		comparator = packedComparator
-	case apis.Distributed:
-		comparator = distributedComparator
+	c.pendingRequests <- req
+
+	select {
+	case res := <-req.response:
+		return res.gs, res.err
+	case <-c.stop:
+		return nil, errors.New("shutting down")
 	}
+}
 
-	allocation, err := c.findReadyGameServerForAllocation(gsa, comparator)
-	if err != nil {
-		return allocation, err
+// runLocalAllocations is a blocking function that runs in a loop
+// looking at c.requestBatches for batches of requests that are coming through.
+func (c *Controller) runLocalAllocations(updateWorkerCount int) {
+	updateQueue := make(chan response)
+	// setup workers for allocation updates
+	c.allocationUpdateWorkers(updateQueue, updateWorkerCount)
```

Comment:
might be more readable to create updateQueue, launch workers and pass the queue to this function.

Reply:
Just to be clear - do you mean, pass the queue to `runLocalAllocations(...)` ?

Or do you mean that `c.allocationUpdateWorkers` should return the upateQueue (which I also quite like - since they are quite coupled together, an we could return an insert only channel from the function - which is tighter encapsulation)

Reply:
I updated this to return the channel, I think this is better, and feel more like a canonical go construct to me.